command/operator_raft_snapshot_inspect.go

```diff
@@ -0,0 +1,651 @@
+	TotalSize    int
+	TotalCountKV int
+}
+
+type MetadataInfo struct {
+	ID      string
+	Size    int64
+	Index   uint64
+	Term    uint64
+	Version raft.SnapshotVersion
+}
+
+type typeStats struct {
+	Name  string
+	Count int
+}
+
+func (c *OperatorRaftSnapshotInspectCommand) Run(args []string) int {
+	flags := c.Flags()
+
+	if err := flags.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	// Fetch snapshot filename
+	var file string
+	args = c.flags.Args()
+
+	switch len(args) {
+	case 0:
+		c.UI.Error("Missing FILE argument")
+		return 1
+	case 1:
+		file = args[0]
+	default:
+		c.UI.Error(fmt.Sprintf("Too many arguments (expected 1, got %d)", len(args)))
+		return 1
+	}
+
+	// Open the file.
+	f, err := os.Open(file)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error opening snapshot file: %s", err))
+		return 1
+	}
+	defer f.Close()
+
+	// Parse metadata and copy state.bin contents to temporary file
+	var readFile *os.File
+	var meta *raft.SnapshotMeta
+	readFile, meta, err = Read(hclog.New(nil), f)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error reading snapshot: %s", err))
+		return 1
+	}
+	defer func() {
+		if err := readFile.Close(); err != nil {
+			c.UI.Error(fmt.Sprintf("Failed to close temp snapshot: %v", err))
+		}
+		if err := os.Remove(readFile.Name()); err != nil {
+			c.UI.Error(fmt.Sprintf("Failed to clean up temp snapshot: %v", err))
+		}
+	}()
+
+	// Parse contents from temporary file
+	info, err := c.enhance(readFile)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error extracting snapshot data: %s", err))
+		return 1
+	}
+
+	formatter, err := NewFormatter(c.flagFormat)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error outputting enhanced snapshot data: %s", err))
+		return 1
+	}
+
+	// Generate structs for the formatter with information we read in
+	metaformat := &MetadataInfo{
+		ID:      meta.ID,
+		Size:    meta.Size,
+		Index:   meta.Index,
+		Term:    meta.Term,
+		Version: meta.Version,
+	}
+
+	// Restructures stats given above to be human readable
+	formattedStatsKV := generateKVStats(info)
+
+	in := &OutputFormat{
+		Meta:         metaformat,
+		StatsKV:      formattedStatsKV,
+		TotalCountKV: info.TotalCountKV,
+	}
+
+	out, err := formatter.Format(in)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	c.UI.Output(out)
+	return 0
+}
+
+func (c *OperatorRaftSnapshotInspectCommand) kvEnhance(val *pb.StorageEntry, info *SnapshotInfo) {
+	if c.kvDetails {
+		if val.Key == "" {
+			return
+		}
+
+		// check for whether a filter is specified. if it is, skip
+		// any keys that don't match.
+		if len(c.kvFilter) > 0 && !strings.HasPrefix(val.Key, c.kvFilter) {
+			return
+		}
+
+		split := strings.Split(string(val.Key), "/")
+
+		// handle the situation where the key is shorter than
+		// the specified depth.
+		actualDepth := c.kvDepth
+		if c.kvDepth > len(split) {
+			actualDepth = len(split)
+		}
+
+		prefix := strings.Join(split[0:actualDepth], "/")
```

Comment:
This panics when `actualDepth` is smaller than 0:

```
[~] vault operator raft snapshot inspect -kvdetails -kvdepth -1 test.snap
panic: runtime error: slice bounds out of range [:-1]

goroutine 34 [running]:
github.com/hashicorp/vault/command.(*OperatorRaftSnapshotInspectCommand).kvEnhance(0x14002ebb620, 0x68?, 0x14002e53db0)
	/Users/jasonodonnell/Git/vault/command/operator_raft_snapshot_inspect.go:233 +0x1a0
github.com/hashicorp/vault/command.(*OperatorRaftSnapshotInspectCommand).enhance.func1(0x14002f0e600?)
	/Users/jasonodonnell/Git/vault/command/operator_raft_snapshot_inspect.go:253 +0x2c
github.com/hashicorp/vault/command.ReadSnapshot.func1()
	/Users/jasonodonnell/Git/vault/command/operator_raft_snapshot_inspect.go:290 +0x58
created by github.com/hashicorp/vault/command.ReadSnapshot in goroutine 1
	/Users/jasonodonnell/Git/vault/command/operator_raft_snapshot_inspect.go:275 +0x168
```

Reply:
good catch!